odoo/addons/base/models/ir_actions_report.py

```diff
@@ -741,27 +741,32 @@ def close_streams(streams):
                                 break
                             node = node['/Next']
                         outlines_pages = sorted(set(outlines_pages))
-                        # There should be only one top-level heading by document
-                        assert len(outlines_pages) == len(res_ids)
-                        # There should be a top-level heading on first page
-                        assert outlines_pages[0] == 0
-                        for i, num in enumerate(outlines_pages):
-                            to = outlines_pages[i + 1] if i + 1 < len(outlines_pages) else reader.numPages
-                            attachment_writer = PdfFileWriter()
-                            for j in range(num, to):
-                                attachment_writer.addPage(reader.getPage(j))
-                            stream = io.BytesIO()
-                            attachment_writer.write(stream)
-                            if res_ids[i] and res_ids[i] not in save_in_attachment:
-                                new_stream = self._postprocess_pdf_report(record_map[res_ids[i]], stream)
-                                # If the buffer has been modified, mark the old buffer to be closed as well.
-                                if new_stream and new_stream != stream:
-                                    close_streams([stream])
-                                    stream = new_stream
-                            streams.append(stream)
-                        close_streams([pdf_content_stream])
+                        if len(outlines_pages) == len(res_ids):
```

Comment:
maybe we could have:

```py
outlines_pages = []
if '/Outlines' in root and '/First' in root['/Outlines']:
    outlines_pages = []
    node = root['/Outlines']['/First']
    while True:
        outlines_pages.append(root['/Dests'][node['/Dest']][0])
        if '/Next' not in node:
            break
        node = node['/Next']
    outlines_pages = sorted(set(outlines_pages))
if len(outlines_pages) == len(res_ids) and outlines_pages[0] == 0:
```
and not duplicate the else (in this branch `len(res_ids) > 1`)